mail_activity_board/views/mail_activity_view.xml

```diff
@@ -0,0 +1,319 @@
+<?xml version="1.0" encoding="utf-8" ?>
+<odoo>
+    <!--
+    VIEWS
+-->
+
+    <!-- FORM POP
+
+        Add a link to the related model instance to ease navigation
+    -->
+    <record id="mail_activity_view_form_popup" model="ir.ui.view">
+        <field
+            name="name"
+        >mail.activity.view.form.popup (in mail_activity_board)</field>
+        <field name="model">mail.activity</field>
+        <field name="inherit_id" ref="mail.mail_activity_view_form_popup" />
+        <field name="arch" type="xml">
+            <field name="activity_type_id" position="before">
+                <field
+                    name="related_model_instance"
+                    invisible="not related_model_instance"
+                />
+            </field>
+        </field>
+    </record>
+
+    <!-- CALENDAR VIEW
+        Add a link to the related model instance to ease navigation
+    -->
+    <record id="mail_activity_view_calendar" model="ir.ui.view">
+        <field
+            name="name"
+        >mail.activity.view.form.calendar (in mail_activity_board)</field>
+        <field name="model">mail.activity</field>
+        <field name="inherit_id" ref="mail.mail_activity_view_calendar" />
+        <field name="arch" type="xml">
+            <field name="res_name" position="before">
+                <field name="related_model_instance" />
+            </field>
+            <field name="res_name" position="attributes">
+                <attribute name="invisible">1</attribute>
+            </field>
+        </field>
+    </record>
+
+    <!-- FORM VIEW -->
+    <record id="mail_activity_view_form_board" model="ir.ui.view">
+        <field name="name">mail.activity.boards.view.form</field>
+        <field name="model">mail.activity</field>
+        <field name="priority">30</field>
+        <field name="arch" type="xml">
+            <form string="Activity Form" create="false" edit="false" delete="false">
+                <sheet string="Activity">
+                    <button
+                        name="open_origin"
+                        type="object"
+                        class="centre oe_link"
+                        nolabel="1"
+                    >
+                        <field name="res_name" invisible="1" />
+                        <h1>
+                            <field name="related_model_instance" />
```

Comment:
You could add: `options="{'no_open': True}" ` this way function open_origin will be called as the button intended